imx/api_error.go

```diff
@@ -0,0 +1,32 @@
+package imx
+
+import (
+	"fmt"
+	"net/http"
+)
+
+type APIError struct {
+	httpResponse *http.Response
+	err          error
+}
+
+/*
+NewAPIError returns an APIError when Api Request fails.
```

Comment:
In TS Core SDK we throw IMXError with domain-specific code and message. No http status code is exposed as it's not as informative for the client. https://github.com/immutable/imx-core-sdk/pull/172

Reply:
Will be addressed by https://immutable.atlassian.net/browse/DX-1242